### PR TITLE
res_pjsip_session: Send Session Interval too small response

### PR DESCRIPTION
**Pull request by btriller**
Handle session interval lower than endpoint's configured minimum timer when sending first answer. Timer setting is checked during this step and needs to handled appropriately. Otherwise no response will be sent at all.

### CODE REVIEW

**Comment by btriller** [comment]
cherry-pick-to: 18
cherry-pick-to: 20
cherry-pick-to: 21

File: res/res_pjsip_session.c [hunk 1]
```diff
@@ -4053,15 +4053,21 @@ static int new_invite(struct new_invite *invite)
 	 * so let's go ahead and send a 100 Trying out to stop any
 	 * retransmissions.
 	 */
+	if (pjsip_inv_initial_answer(invite->session->inv_session, invite->rdata, 100, NULL, NULL, &tdata) != PJ_SUCCESS) {
+		if (tdata == NULL) {
```
Comment (jcolp) [inline res/res_pjsip_session.c:4057]: This needs to be updated according to [coding guidelines](https://docs.asterisk.org/Development/Policies-and-Procedures/Historical-Policies-and-Procedures/Code-Review/Coding-Guidelines/).

if (!tdata) {
} else {
}

**Review by jcolp — CHANGES_REQUESTED** [review]
I think the commit message should be updated to include what previously happened, and what now happens, from a SIP response perspective in this scenario.

File: res/res_pjsip_session.c (hunk 1 above)
Comment (jcolp) [reply res/res_pjsip_session.c:4057]: if (!tdata) {

**Review by jcolp — APPROVED** [review]

**Pull request merged**